### PR TITLE
Override Comparison Operators in AccessCompiler

### PR DESCRIPTION
**Pull request by the-wondersmith**
The PR fixes Issue #6

### CODE REVIEW

File: sqlalchemy_access/base.py [hunk 1]
```diff
@@ -13,11 +13,20 @@
...
+from sqlalchemy.sql.operators import ne, is_, isnot
 from sqlalchemy.engine import default, base, reflection
 from sqlalchemy import processors
 
 import pyodbc
 
+# These operator overrides are required in order to make
+# the AccessCompiler class correctly format emmited queries
+OPERATORS[ne] =  " <> "
+OPERATORS[is_] =  " = "
```
Comment (gordthompson) [inline sqlalchemy_access/base.py]: My understanding is that `is_` and `isnot_` are normally used for handling nulls (e.g., `... WHERE foo IS NULL`). Do you have a use-case where they should really should render ` = `  and ` <> ` instead of ` IS ` and ` IS NOT `?

Comment (the-wondersmith) [reply sqlalchemy_access/base.py]: I wasn't aware of that in particular, but in my testing it appears that Access is finicky about IS comparisons to True and False, but isn't finicky about the same comparisons using = .

The behavior on the Access side of things appears to be identical if you use = for comparisons to True, False, and NULL. Access throws an error when I try to use IS for anything *but* NULL. Given that, it comes down to a code style thing I think. It seems more Pythonic to my eye to use `.is_(True)` rather than ` == True `

Forgoing the override causes SQLAlchemy to throw an exception when trying to do things like `session.query(Sale).filter(Sale.taxable.is_(True)).all()`, which forces you to use `session.query(Sale).filter(Sale.taxable == True).all()` or `session.query(Sale).filter(Sale.taxable == 1).all()` (both of which appear to work, but just aren't quite as Pythic feeling.)

Making the change still allows the use of ` == ` on the Python side of things, it just *also* allows the use of ` .is_() ` and ` .isnot_() ` to function as expected. With no negative or restrictive impact on usability, it seemed silly not to include it.

Comment (gordthompson) [reply sqlalchemy_access/base.py]: Access "Yes/No" columns are a bit of a special case because they are not nullable. If we try to assign a `NULL` value to that column it is written to the database as `False`. Thus, `WHERE colname = NULL` and `WHERE colname IS NULL` will both always return zero rows. However, for a nullable column type like "Short Text" `WHERE colname IS NULL` will return the rows where the column really is NULL, while `WHERE colname = NULL` will always return zero rows.

TL;DR: `is` and equality really do mean different things, both in SQL and [in Python](https://stackoverflow.com/q/132988/2144390).

Comment (the-wondersmith) [reply sqlalchemy_access/base.py]: Ah. Well, thank you for clearing that up for me. I'll edit the patch.

**Comment by gordthompson** [comment]
Done. Thanks!

File: sqlalchemy_access/base.py [hunk 2]
```diff
@@ -13,11 +13,18 @@
...
+from sqlalchemy.sql.compiler import OPERATORS
+from sqlalchemy.sql.operators import ne, is_, isnot
 from sqlalchemy.engine import default, base, reflection
 from sqlalchemy import processors
 
 import pyodbc
 
+# The != operator override is required in order to make
+# the AccessCompiler class correctly format emitted queries
+OPERATORS[ne] =  " <> "
```
Comment (CaselIT) [inline sqlalchemy_access/base.py:25]: won't this override the global operator in sqlalchemy and create problem if more than one dialect is used?

maybe adding a function to the compiler `def visit_ne_binary(self, binary, operator, **kw):` would be better

Comment (gordthompson) [reply sqlalchemy_access/base.py:25]: Good point. I'll have to take a look at that.

Comment (gordthompson) [reply sqlalchemy_access/base.py:25]: Turns out that the "Big 5" (internal) dialects + Firebird all support both `!=` and `<>` for "not equals" so this particular change is not a show-stopper. I have implemented your suggestion and pushed it to master; it will be included in the next release. Thanks!
